adding dart language (#56)
devstrons/hello-world: lang/dart.md

---
lang: Dart
syntax: dart
title: Dart is Smart Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

> ## What is Dart?

Reviewer: As @vsnthdev said, we should add an extra line here. 🙂
```suggestion
## What is Dart?

```

Reviewer 2: @Panquesito7: Yes! Thanks to your catchy evil eye! : )

- [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, `and` JavaScript. `If` you know any of these programming languages, you can easily learn the Dart programming language.

- [Dart](https://dart.dev) is an open-source programming language that is widely used to develop the mobile application, modern web applications, desktop applications, `and` the Internet of Things (IoT) using by Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, refield generics, `and` type  interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [dart tutorials](https://dart.dev/tutorials).

> ## What is Dart used for?

Reviewer: Same here.
```suggestion
## What is Dart used for?

```

Reviewer 2: @chandan0000: After every `##` sentence, left a line below!

- **[OverReact](https://workiva.github.io/over_react)** - A library for building statically-typed React UI components.
- **[VueDart](https://vuedart.dev)** - Create Vue web apps using Dart, a progressive, incrementally-adoptable framework for building Web UIs.

**Server Frameworks:**
- **[Jaguar](https://github.com/Jaguar-dart/jaguar)** - A server framework built for speed, simplicity and extensibility.
- **[Start](https://github.com/lvivski/start)** - Sinatra inspired web framework to serve static files, handle dynamic requests, websockets and create JSON responses.
- **[Shelf](https://pub.dev/packages/shelf)** - Shelf makes it easy to create and compose web servers and parts of web servers.
- **[Vane](https://github.com/Scorpiion/Vane)** - Framework with built-in server runtime environment and middleware system.
- **[Rikulo Stream](https://github.com/rikulo/stream)** - Lightweight web server with request routing, filtering, template engine, WebSocket, MVC design pattern, and file-based static resources.
- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
 

> ## Why to use Dart?

Reviewer: Same here.
```suggestion
## Why use Dart?

```

Reviewer 2: @Panquesito7: Sorry, it's still there, likewise, I've to mark it resolved! : )

Reviewer: Updated, forgot that 😅 it got updated by Grammarly.

- Dart is really flexible
- Dart embraced the open source ecosystem
- Dart is pretty easy to learn
- Dart is a good first language
- Dart has great tooling support
- Dart is a robust language
- Dart ensures productivity
- Dart is used by Flutter
- Dart is backed by a tech giant
- Dart solved a lot of problems, efficiently

> ## Tools You Will Need

Reviewer: Same here.
```suggestion
## Tools You Will Need

```

For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
You will also need the following softwares :
- Linux, macOS or Windows XP/7/8/10 operating system
- Dart SDK
- Text Editors

> ## Setting up Dart development environment

Reviewer: And here.
```suggestion
## Setting up Dart development environment

```

Install using apt-get
Install a Debian package
Installation using apt-get

**Step 1:**

 Type the following commands for a one-time setup.

```bash
sudo apt-get update  
sudo apt-get install apt-transport-https  
sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
```

**Step 2:**

 Type the following command in terminal to install the Dart SDK using apt-get option.

```bash
sudo apt-get update  
sudo apt-get install dart
```

It will successfully download the Dart SDK.

**Installation a Debian Package:**

We can download Dart SDK as a Debian package in the .deb package format. To make all Dart binaries accessible, we have to change the PATH by typing the following command.

```bash
export PATH="$PATH:/usr/lib/dart/bin"
```

To change the PATH for upcoming terminal sessions, use a below command:

```bash
echo 'export PATH="$PATH:/usr/lib/dart/bin"' >> ~/.profile
```

> ## Popular Dart IDEs/editors 

Reviewer: Here too.
```suggestion
## Popular Dart IDEs/editors

```

---
lang: Dart
syntax: dart
title: Dart is Smart Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

## What is Dart?

> - Dart is a dynamic, class-based, object-oriented programming language with closure and lexical scope. Syntactically, it is quite similar to Java, C, and JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.

Reviewer 2: ```suggestion
- [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure and lexical scope. Syntactically, it is quite similar to Java, C, and JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.
```

> - Dart is an open-source programming language which is widely used to develop the mobile application, modern web-applications, desktop application, and the Internet of Things (IoT) using by Flatter framework. It also supports a few advance concepts such as interfaces, mixins, abstract classes, refield generics, and type interface. It is a compiled language and supports two types of compilation techniques. [Click Here Dart official Website:](https://dart.dev/tutorials)

Reviewer 2: ```suggestion
- Dart is an open-source programming language that is widely used to develop the mobile application, modern web applications, desktop applications, and the Internet of Things (IoT) using by Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, refield generics, and type interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [dart tutorials](https://dart.dev/tutorials).
```

- **[Shelf](https://pub.dev/packages/shelf)** - Shelf makes it easy to create and compose web servers and parts of web servers.
- **[Vane](https://github.com/Scorpiion/Vane)** - Framework with built-in server runtime environment and middleware system.
- **[Rikulo Stream](https://github.com/rikulo/stream)** - Lightweight web server with request routing, filtering, template engine, WebSocket, MVC design pattern, and file-based static resources.
- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
 
## Why to use Dart?
- Dart is really flexible
- Dart embraced the open source ecosystem
- Dart is pretty easy to learn
- Dart is a good first language
- Dart has great tooling support
- Dart is a robust language
- Dart ensures productivity
- Dart is used by Flutter
- Dart is backed by a tech giant
- Dart solved a lot of problems, efficiently

## Tools You Will Need
For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
You will also need the following softwares :

> - Linux, macOS or Windows xp/7/8/10 operating system

Reviewer 2: ```suggestion
- Linux, macOS or Windows XP/7/8/10 operating system
```

- Dart SDK

> - Text Editors

Reviewer 2: What kind of Text Editors, beginners will use? :thinking: 
Pl mention! : )

Reviewer 3: Most people will use:
1. Sublime Text
2. Visual Studio Code
3. Notepad++
4. Atom

Reviewer 2: @chandan0000: Yep, agree with Vasanth, did you give a check to include such stuff with more links to reference?

Download Dart SDK
Click on the Dart SDK link. The download will start immediately and a zip file will be downloaded.

**Step 2:** Extract zip file
Extract the contents of Dart SDK zip file. The contents of the folder would be as shown below.

**Step 3:**
Add Dart Path to PATH Environment Variable
As of now, you can run dart command only from bin folder of dart sdk. To run dart command from anywhere in your file system, add dart bin path to PATH environment variable.
Open Environment Variables. Under System variables, click on Path and click Edit button.
Edit environment variable window appears. Click on New and paste the dart sdk bin path as shown below.

**Step 4:** Restart Command Prompt
Close the existing command prompt window and open a new command prompt.

Just run the command: 

```bash  
dart --version 
```

**Install the Dart SDK on Linux:**
The steps of Dart installation on Linux is given below.
Before installing the Dart, if you are Debian/Ubuntu on AMD64(64-bit Intel) in your local machine, you can install the Dart through one of the following options.

Install using apt-get
Install a Debian package
Installation using apt-get

> **Step -1:**

Reviewer 2: ```suggestion
**Step 1:**
```

 Type the following commands for a one-time setup.

```bash
sudo apt-get update  
sudo apt-get install apt-transport-https  
sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
```

> **Step - 2:**

Reviewer 2: ```suggestion
**Step 2:**
```

 Type the following command in terminal to install the Dart SDK using apt-get option.

```bash
sudo apt-get update  
sudo apt-get install dart
```

It will successfully download the Dart SDK.

**Installation a Debian Package:**

We can download Dart SDK as a Debian package in the .deb package format. To make all Dart binaries accessible, we have to change the PATH by typing the following command.

```bash
export PATH="$PATH:/usr/lib/dart/bin"
```

To change the PATH for upcoming terminal sessions, use a below command:

```bash
echo 'export PATH="$PATH:/usr/lib/dart/bin"' >> ~/.profile
```

## Popular Dart IDEs/editors 
- The Eclipse,VsCode, IntelliJ, and WebStorm are the IDEs from the Jet brains that support the Dart Programming, but WebStorm is more popular than others. We can download it from <https://www.jetbrains.com/webstorm/download/>.

## Some Cool projects in Dart

**1. [Invoice Naija](https://github.com/invoiceninja/admin-portal) :** This is a business application for documenting invoices, tasks, and expenses.
It was built using flutter and laravel.
- Invoice Naija
- Extra Features
- Clients
- Products
- Quotes
- Payments
 

> 

Reviewer 2: Reduce line -> 147 backspace!
Single `enter` is enough to render it!

- Further, Ability to add individual assets in addition to wallets

**3 [Weight Tracker](https://github.com/MSzalek-Mobile/weight_tracker) :** Weight Tracker is an application dedicated to people who want to dump or maintain weight. It allows you to keep an eye on your progress and pursue your weight goals.
- **Screenshot Image**
- **Extra Features**
- **Simple**
- **User-Friendly** 

**4. [Flutter Music Player](https://github.com/iampawan/Flutter-Music-Player) :**
Flutter Music Player is an open-source music player application, available only on Andriod, for now.
- Screenshot Image
- Extra Features
- Fetch local songs (Inbuilt Permission)
- Beautiful UI with multiple themes(Coming Soon)
- Animation
- Full Fledged Example
- play (local files)
- stop
- pause
- seek
- shuffle
- Album Art
- onComplete
- onDuration / onCurrentPosition

**5. [Trovami](https://github.com/Samaritan1011001/Trovami) :**
Trovami is a flutter based android application, you can use to share your current location with other persons.
- preview
- Extra Features
- Create groups and choose to share your live location
- View group members on the map
- Google Sign-in
 

> **Thanks For Reading**

Reviewer 2: ```suggestion
**Thanks For Reading!**
```

Reviewer 3: It really should be 👇 

```
**Thanks for reading!**
```

I cannot see who you've used title case here 😅

---
lang: Dart
syntax: dart

> title: Dart is Smart Language!

Reviewer 2: ```suggestion
title: Dart is Sky Language!
```

author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

> ## 🎓 What is Dart?

Reviewer 2: ```suggestion
## 🐦 What is Dart?
```

- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
 
## 🤔 Why to use Dart?

- Dart is really flexible
- Dart embraced the open source ecosystem
- Dart is pretty easy to learn
- Dart is a good first language
- Dart has great tooling support
- Dart is a robust language
- Dart ensures productivity
- Dart is used by Flutter
- Dart is backed by a tech giant
- Dart solved a lot of problems, efficiently

## 🛠️ Tools You Will Need

For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
You will also need the following softwares :
- Linux, macOS or Windows XP/7/8/10 operating system
- Dart SDK
- Text Editors
## ⚙️ Setting up Dart development environment

**Executing Script Online with DartPad:**
You may test your scripts online by using the online editor at <https://dartpad.dartlang.org/>. The Dart Editor executes the script and displays both HTML as well as console output. The online editor is shipped with a set of preset code samples.

> **Dart Installation:**

Reviewer 2: Left a between b/wn. each & every sentence i.e, topics/subtopics!

**Install Dart on Windows:**
In this tutorial, we will provide a step by step process to get Dart installed on Windows.

Dart SDK comes as a pre-compiled version. So, download and extracting is all you need to do.

**Step 1:** Download Dart SDK
Go to Dart SDK archive page.
The URL is [Click Here ](https://dart.dev/tools/sdk/archive)
Download Dart SDK
Click on the Dart SDK link. The download will start immediately and a zip file will be downloaded.

**Step 2:** Extract zip file
Extract the contents of Dart SDK zip file. The contents of the folder would be as shown below.

**Step 3:**
Add Dart Path to PATH Environment Variable
As of now, you can run dart command only from bin folder of dart sdk. To run dart command from anywhere in your file system, add dart bin path to PATH environment variable.
Open Environment Variables. Under System variables, click on Path and click Edit button.
Edit environment variable window appears. Click on New and paste the dart sdk bin path as shown below.

**Step 4:** Restart Command Prompt
Close the existing command prompt window and open a new command prompt.

Just run the command: 

```bash  
dart --version 
```

> ** ⚙️ Install the Dart SDK on Linux:**

Reviewer 2: ```suggestion
#### **⚙️ Install the Dart SDK on Linux:**
```

The steps of Dart installation on Linux is given below.
Before installing the Dart, if you are Debian/Ubuntu on AMD64(64-bit Intel) in your local machine, you can install the Dart through one of the following options.

Install using apt-get
Install a Debian package
Installation using apt-get

**Step 1:**

 Type the following commands for a one-time setup.

```bash
sudo apt-get update  
sudo apt-get install apt-transport-https  
sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
```

**Step 2:**

 Type the following command in terminal to install the Dart SDK using apt-get option.

```bash
sudo apt-get update  
sudo apt-get install dart
```

It will successfully download the Dart SDK.

> ** ⚙️ Installation a Debian Package:**

Reviewer 2: ```suggestion
#### **⚙️ Installation a Debian Package:**
```

---
lang: Dart
syntax: dart
title: Dart is Smart Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

## What is Dart?

> - [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, `and` JavaScript. `If` you know any of these programming languages, you can easily learn the Dart programming language.

Reviewer: Same here: minor spelling improvements.
```suggestion
- [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, `and` JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.
```

> - [Dart](https://dart.dev) is an open-source programming language that is widely used to develop the mobile application, modern web applications, desktop applications, `and` the Internet of Things (IoT) using by Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, refield generics, `and` type  interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [dart tutorials](https://dart.dev/tutorials).

Reviewer: Minor grammar improvements.
```suggestion
- [Dart](https://dart.dev) is an open-source programming language that is widely used to develop mobile applications, modern web applications, desktop applications, and the Internet of Things (IoT) using the Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, field generics, and type interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [Dart tutorials](https://dart.dev/tutorials).
```

> ## What is Dart used for?

Reviewer: This is still not fixed (missing an extra line). 🙂
```suggestion
## What is Dart used for?

```

---
lang: Dart
syntax: dart
title: Dart is Sky Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

## 🐦 What is Dart?

- [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, `and` JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.

- [Dart](https://dart.dev) is an open-source programming language that is widely used to develop mobile applications, modern web applications, desktop applications, and the Internet of Things (IoT) using the Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, field generics, and type interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [Dart tutorials](https://dart.dev/tutorials).

> ## 📚 What is Dart used for?

Reviewer: Still needs an extra space here.
```suggestion
## 📚 What is Dart used for?

```

Download Dart SDK
Click on the Dart SDK link. The download will start immediately and a zip file will be downloaded.

**Step 2:** Extract zip file
Extract the contents of Dart SDK zip file. The contents of the folder would be as shown below.

**Step 3:**
Add Dart Path to PATH Environment Variable
As of now, you can run dart command only from bin folder of dart sdk. To run dart command from anywhere in your file system, add dart bin path to PATH environment variable.
Open Environment Variables. Under System variables, click on Path and click Edit button.
Edit environment variable window appears. Click on New and paste the dart sdk bin path as shown below.

**Step 4:** Restart Command Prompt
Close the existing command prompt window and open a new command prompt.
Just run the command: 

```bash  
dart --version 
```
#### **⚙️ Install the Dart SDK on Linux:**

The steps of Dart installation on Linux is given below.
Before installing the Dart, if you are Debian/Ubuntu on AMD64(64-bit Intel) in your local machine, you can install the Dart through one of the following options.
Install using apt-get
Install a Debian package
Installation using apt-get

**Step 1:**

>  Type the following commands for a one-time setup.

Reviewer: There's a whitespace line here that should be removed.
```suggestion
Type the following commands for a one-time setup.
```

```bash
sudo apt-get update  
sudo apt-get install apt-transport-https  
sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
```

**Step 2:**

 Type the following command in terminal to install the Dart SDK using apt-get option.

```bash
sudo apt-get update  
sudo apt-get install dart
```

It will successfully download the Dart SDK.

#### **⚙️ Installation a Debian Package:**

We can download Dart SDK as a Debian package in the .deb package format. To make all Dart binaries accessible, we have to change the PATH by typing the following command.

```bash
export PATH="$PATH:/usr/lib/dart/bin"
```

To change the PATH for upcoming terminal sessions, use a below command:
```bash
echo 'export PATH="$PATH:/usr/lib/dart/bin"' >> ~/.profile
```

#### **⚙️Install the Dark SDK on Mac**

> **Step 1:** We should have Homebrew package manager, but if we don't have it then install the Homebrew and run the following command. It will successfully download the Dart on the Mac.

Reviewer: Minor spelling typo. Also needs extra space before the code backticks.
```suggestion
**Step 1:** We should have a Homebrew package manager, but if we don't have it then install the Homebrew and run the following command. It will successfully download the Dart on the Mac.
```

```bash
 brew tap dart-lang/dart  
 brew install dart  
```

> **Step 2:** To verify which version we have installed, use the following command.

Reviewer: Same here: needs extra space before code backticks.
```suggestion
**Step 2:** To verify which version we have installed, use the following command.

```

---
lang: Dart
syntax: dart
title: Dart is Smart Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

> ## What is Dart?

Reviewer 3: Add a newline here, otherwise, the frontend could break.

- Dart is a dynamic, class-based, object-oriented programming language with closure and lexical scope. Syntactically, it is quite similar to Java, C, and JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.

> - Dart is an open-source programming language which is widely used to develop the mobile application, modern web-applications, desktop application, and the Internet of Things (IoT) using by Flatter framework. It also supports a few advance concepts such as interfaces, mixins, abstract classes, refield generics, and type interface. It is a compiled language and supports two types of compilation techniques. **[Dart official Website: ](https://dart.dev/tutorials)** https://dart.dev/tutorials

Reviewer 3: Remove the extra space at the end as it messes with the markdown renderer.
Change `[Dart official Website: ]` to 👇 

```
[Dart official Website:](https://dart.dev/tutorials))
```

## What is Dart used for?
Dart programming language can be used for web and mobile applications and is also suitable for server and desktop applications.
Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, also Google had all the control about this language.

>   **Client Web App Frameworks:**

Reviewer 3: Markdown doesn't respect empty spaces at the beginning of the lines. So, it would be more uniform and readable if you can trim extra spaces. Replace line 22 with 👇 

```
**Client Web App Frameworks:**
```

- **[AngularDart](https://github.com/angulardart/angular)** - AngularDart is a development platform for building mobile and desktop web applications.

- **[Flutter](https://flutter.dev/)** - Flutter is a framework to build high-performance, cross-platform mobile apps, allowing applications to be written for Android, iOS and Web Apps.

- **[MDL/Dart](http://mdl.mikemitterer.at)** - Material Design Lite for Dart is a framework of components for web developers based on Google's Material Design philosophy.

- **[OverReact](https://workiva.github.io/over_react)** - A library for building statically-typed React UI components.
- **[VueDart](https://vuedart.dev)** - Create Vue web apps using Dart, a progressive, incrementally-adoptable framework for building Web UIs.

>   **Server Frameworks:**

Reviewer 3: Again, markdown doesn't really respect spaces before the content starts in a new line. Trim the spaces in the beginning.

- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
 
 
## Why to use Dart?
- Dart is really flexible
- Dart embraced the open source ecosystem
- Dart is pretty easy to learn
- Dart is a good first language
- Dart has great tooling support
- Dart is a robust language
- Dart ensures productivity
- Dart is used by Flutter
- Dart is backed by a tech giant
- Dart solved a lot of problems, efficiently

## Tools You Will Need
For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
You will also need the following softwares :
- Linux, macOS or Windows xp/7/8/10 operating system
- Dart SDK
- Text Editors

## Setting up Dart development environment
**Executing Script Online with DartPad:**

You may test your scripts online by using the online editor at https://dartpad.dartlang.org/. The Dart Editor executes the script and displays both HTML as well as console output. The online editor is shipped with a set of preset code samples.

> 

Reviewer 3: Remove unnecessary empty lines, HTML anyhow will only render one empty newline. So, remove it to make it more readable.

**Dart Installation:**
 
 
**Install Dart on Windows:**
In this tutorial, we will provide a step by step process to get Dart installed on Windows.

Dart SDK comes as a pre-compiled version. So, download and extracting is all you need to do.

**Step 1:** Download Dart SDK
Go to Dart SDK archive page.

> The URL is https://dart.dev/tools/sdk/archive.

Reviewer 3: Replace this line with 👇 

```
The URL is [https://dart.dev/tools/sdk/archive](https://dart.dev/tools/sdk/archive).
```

Reviewer: Using `<https://dart.dev/tools/sdk/archive>` is better as it is a bare URL.

**Step 2:** Extract zip file
Extract the contents of Dart SDK zip file. The contents of the folder would be as shown below.

**Step 3:**
Add Dart Path to PATH Environment Variable
As of now, you can run dart command only from bin folder of dart sdk. To run dart command from anywhere in your file system, add dart bin path to PATH environment variable.

Open Environment Variables. Under System variables, click on Path and click Edit button.

Edit environment variable window appears. Click on New and paste the dart sdk bin path as shown below.

**Step 4:** Restart Command Prompt
Close the existing command prompt window and open a new command prompt.

Just run the dart command:  `dart --version`

**Install the Dart SDK on Linux:**

The steps of Dart installation on Linux is given below.

Before installing the Dart, if you are Debian/Ubuntu on AMD64(64-bit Intel) in your local machine, you can install the Dart through one of the following options.

Install using apt-get
Install a Debian package
Installation using apt-get

**Step -1:**

 Type the following commands for a one-time setup.

> $sudo apt-get update  

Reviewer 3: Surround code blocks with 3 backticks to render them as code blocks. Also, specify the syntax to allow us to set proper syntax highlighting.

Example code block:
<pre></code>```bash
sudo apt-get update
```</code></pre>

ℹ️ **Note:** Also, do not specify any elevation marks before the command as we'll be implementing a copy button in the future. Plus it's really inconvenient to copy the command that way... Feel free to contact me on Discord at `Vasanth Srivatsa#2000` on the [DEVSTR😊NS' Discord server](https://discord.gg/52VMn9edNH).

$ sudo apt-get install apt-transport-https  
  
$ sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
  
$ sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
**Step - 2:**

 Type the following command in terminal to install the Dart SDK using apt-get option.

$sudo apt-get update  
$ sudo apt-get install dart  

> It will successfully download the Dart SDK.

Reviewer 3: Leave 1 new line before and after the code blocks in order to not confuse the markdown engine.

**Installation a Debian Package:**

We can download Dart SDK as a Debian package in the .deb package format. To make all Dart binaries accessible, we have to change the PATH by typing the following command.

export PATH="$PATH:/usr/lib/dart/bin"  
To change the PATH for upcoming terminal sessions, use a below command:

$ echo 'export PATH="$PATH:/usr/lib/dart/bin"' >> ~/.profile  

## Popular Dart IDEs/editors 

> - The Eclipse,VsCode, IntelliJ, and WebStorm are the IDEs from the Jet brains that support the Dart Programming, but WebStorm is more popular than others. We can download it from https://www.jetbrains.com/webstorm/download/#section=windows-version.

Reviewer 3: 1. Have spaces between these editors.
2. It's `VSCode` or `Visual Studio Code`.
3. Replace the following 👇 

```
We can download it from https://www.jetbrains.com/webstorm/download/#section=windows-version.
```

with 👇 

```
We can download it from [here](https://www.jetbrains.com/webstorm/download).
```

## Some Cool projects in Dart

- **1. [Invoice Naija](https://github.com/invoiceninja/admin-portal) :** This is a business application for documenting invoices, tasks, and expenses.
It was built using flutter and laravel.

- Invoice Naija
- Extra Features
- Clients
- Products
- Quotes
- Payments
 

**2. [Bit Alarm](https://github.com/damoonrashidi/bitalarm) :** This is a cryptocurrency application created to keep track of different cryptocurrencies

Bit Alarm
Extra Features
Loading indicator.
 Graph out historical data for a currency
Scan QR-code to add a wallet to wallet list
 Remove wallet from the list
 Dynamic portfolio based on the address (ETH + ERC20-tokens, LTC, BTC, BCH, DASH and ADA for now)
 Further, Ability to add individual assets in addition to wallets

**3 [Weight Tracker](https://github.com/MSzalek-Mobile/weight_tracker) :** Weight Tracker is an application dedicated to people who want to dump or maintain weight. It allows you to keep an eye on your progress and pursue your weight goals.

- **Screenshot Image**

- **Extra Features**

> 

Reviewer 3: Remove newlines in between here. As they won't be rendered anyways. Remove to improve readability.

- **Simple**

- **User-Friendly** 

**4. [Flutter Music Player](https://github.com/iampawan/Flutter-Music-Player) :**
Flutter Music Player is an open-source music player application, available only on Andriod, for now.

- Screenshot Image
- Extra Features
- Fetch local songs (Inbuilt Permission)
- Beautiful UI with multiple themes(Coming Soon)
- Animation
- Full Fledged Example
- play (local files)
- stop
- pause
- seek
- shuffle
- Album Art
- onComplete
- onDuration / onCurrentPosition

**5. [Trovami](https://github.com/Samaritan1011001/Trovami) :**
Trovami is a flutter based android application, you can use to share your current location with other persons.

- preview
- Extra Features
- Create groups and choose to share your live location
- View group members on the map

> - Google Sign-in

Reviewer 3: 🤔 maybe a nice _thank you for reading_ the message? 🤷‍♂️

- Further, Ability to add individual assets in addition to wallets

**3 [Weight Tracker](https://github.com/MSzalek-Mobile/weight_tracker) :** Weight Tracker is an application dedicated to people who want to dump or maintain weight. It allows you to keep an eye on your progress and pursue your weight goals.
- **Screenshot Image**
- **Extra Features**
- **Simple**
- **User-Friendly** 

**4. [Flutter Music Player](https://github.com/iampawan/Flutter-Music-Player) :**
Flutter Music Player is an open-source music player application, available only on Andriod, for now.
- Screenshot Image
- Extra Features
- Fetch local songs (Inbuilt Permission)
- Beautiful UI with multiple themes(Coming Soon)
- Animation
- Full Fledged Example
- play (local files)
- stop
- pause
- seek
- shuffle
- Album Art
- onComplete
- onDuration / onCurrentPosition

**5. [Trovami](https://github.com/Samaritan1011001/Trovami) :**
Trovami is a flutter based android application, you can use to share your current location with other persons.
- preview
- Extra Features
- Create groups and choose to share your live location
- View group members on the map
- Google Sign-in

> ## ✨Conclusion

Reviewer: Missing a space.
```suggestion
## ✨Conclusion

```

---
lang: Dart
syntax: dart
title: Dart is Sky Language!
author: chandan0000
---

```dart
void main(){
    print("hello world");
}
```

## 🐦 What is Dart?

> - [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, `and` JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.

Reviewer: ```suggestion
- [Dart](https://dart.dev) is a dynamic, class-based, object-oriented programming language with closure `and` lexical scope. Syntactically, it is quite similar to Java, C, and JavaScript. If you know any of these programming languages, you can easily learn the Dart programming language.
```

- [Dart](https://dart.dev) is an open-source programming language that is widely used to develop mobile applications, modern web applications, desktop applications, and the Internet of Things (IoT) using the Flatter framework. It also supports a few advanced concepts such as interfaces, mixins, abstract classes, field generics, and type interfaces. It is a compiled language and supports two types of compilation techniques. To explore & learn, redirect to [Dart tutorials](https://dart.dev/tutorials).

## 📚 What is Dart used for?

> Dart programming language can be used for web and mobile applications and is also suitable for server and desktop applications.

Reviewer: ```suggestion
The Dart programming language can be used for web and mobile applications and is also suitable for server and desktop applications.
```

> Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, also Google had all the control about this language.

Reviewer: ```suggestion
Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, plus Google had all the control about this language.
```

> Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, also Google had all the control about this language.

Reviewer: I think `plus` looks better than `also`.
```suggestion
Dart wasn’t popular and was known as an orphan, but it had a compiler for a Dart VM, plus Google had all the control about this language.
```

- **[Shelf](https://pub.dev/packages/shelf)** - Shelf makes it easy to create and compose web servers and parts of web servers.
- **[Vane](https://github.com/Scorpiion/Vane)** - Framework with built-in server runtime environment and middleware system.
- **[Rikulo Stream](https://github.com/rikulo/stream)** - Lightweight web server with request routing, filtering, template engine, WebSocket, MVC design pattern, and file-based static resources.
- **[Alfred]( https://github.com/rknell/alfred)** - Minimal ExpressJS-inspired server with routing and middlware.
 
## 🤔 Why to use Dart?

- Dart is really flexible
- Dart embraced the open source ecosystem
- Dart is pretty easy to learn
- Dart is a good first language
- Dart has great tooling support
- Dart is a robust language
- Dart ensures productivity
- Dart is used by Flutter
- Dart is backed by a tech giant
- Dart solved a lot of problems, efficiently

## 🛠️ Tools You Will Need

For setting up environment for performing java codes you need with a minimum of 2 GB of RAM.
You will also need the following softwares :

> - Linux, macOS or Windows XP/7/8/10 operating system

Reviewer: ```suggestion
- Linux, macOS, or Windows XP/7/8/10 operating system
```

Open Environment Variables. Under System variables, click on Path and click Edit button.
Edit environment variable window appears. Click on New and paste the dart sdk bin path as shown below.

**Step 4:** Restart Command Prompt
Close the existing command prompt window and open a new command prompt.
Just run the command: 

```bash  
dart --version 
```
#### **⚙️ Install the Dart SDK on Linux:**

The steps of Dart installation on Linux is given below.
Before installing the Dart, if you are Debian/Ubuntu on AMD64(64-bit Intel) in your local machine, you can install the Dart through one of the following options.
Install using apt-get
Install a Debian package
Installation using apt-get

**Step 1:**
Type the following commands for a one-time setup.

```bash
sudo apt-get update  
sudo apt-get install apt-transport-https  
sudo sh -c 'wget -qO- https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -'  
sudo sh -c 'wget -qO- https://storage.googleapis.com/download.dartlang.org/linux/debian/dart_stable.list > /etc/apt/sources.list.d/dart_stable.list'  
```

**Step 2:**

>  Type the following command in terminal to install the Dart SDK using apt-get option.

Reviewer: ```suggestion
Type the following command in the terminal to install the Dart SDK using the `apt-get` option.
```

```

**Step 2:**

Type the following command in the terminal to install the Dart SDK using the `apt-get` option.

```bash
sudo apt-get update  
sudo apt-get install dart
```

It will successfully download the Dart SDK.

#### **⚙️ Installation a Debian Package:**

We can download Dart SDK as a Debian package in the .deb package format. To make all Dart binaries accessible, we have to change the PATH by typing the following command.

```bash
export PATH="$PATH:/usr/lib/dart/bin"
```

To change the PATH for upcoming terminal sessions, use a below command:

```bash
echo 'export PATH="$PATH:/usr/lib/dart/bin"' >> ~/.profile
```

#### **⚙️ Install the Dark SDK on Mac**

**Step 1:** We should have a Homebrew package manager, but if we don't have it then install the Homebrew and run the following command. It will successfully download the Dart on the Mac.

```bash
 brew tap dart-lang/dart  
 brew install dart  
```

**Step 2:** To verify which version we have installed, use the following command.

```bash
brew info dart
```

## 💻 Popular Dart IDEs/editors

> The Eclipse,VsCode, IntelliJ, and WebStorm are the IDEs from the Jet brains that support the Dart Programming, but WebStorm is more popular than others. We can download it from <https://www.jetbrains.com/webstorm/download/>.

Reviewer: ```suggestion
The Eclipse, VSCode, IntelliJ, and WebStorm are the IDEs from the Jet brains that support the Dart Programming, but WebStorm is more popular than others. We can download it from <https://www.jetbrains.com/webstorm/download/>.
```

Reviewer 2: @Panquesito7: You can edit everything in one PR, whatever you want to change in any language. : )
